### PR TITLE
Add LocalesMenuButton

### PR DESCRIPTION
**Pull request by djhi**
I'm considering renaming this component to `LocaleMenuButton`. @fzaninotto, what do you think ?

### CODE REVIEW

**Review by fzaninotto — APPROVED** [review]

File: packages/ra-ui-materialui/src/button/LocalesMenuButton.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { Basic } from './LocalesMenuButton.stories';
+
+describe('LanguageSwitcher', () => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/LocalesMenuButton.spec.tsx:5]:
```suggestion
describe('LocalesMenuButton', () => {
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/LocalesMenuButton.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import {
+    Admin,
+    Resource,
+    LocalesMenuButton,
+    List,
+    SimpleList,
+    Layout,
+    AppBar,
+} from 'react-admin';
+import { Box, Typography } from '@material-ui/core';
```
Comment (WiXSL) [inline docs/LocalesMenuButton.md:26]:
```suggestion
import { Box, Typography } from '@mui/material';
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/button/LocalesMenuButton.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { Basic } from './LocalesMenuButton.stories';
+
+describe('LocalesMenuButton', () => {
+    test('should allow to change language', async () => {
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/button/LocalesMenuButton.spec.tsx:6]:
```suggestion
    it('should allow to change language', async () => {
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/LocalesMenuButton.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+[`LocalesMenuButton`]: https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/button/LocalesMenuButton.tsx
+
+## `sx`: CSS API
+
+The `<LocalesMenuButton>` component accepts the usual `className` prop. You can also override many styles of the inner components thanks to the `sx` property (as most MUI components, see their [documentation about it](https://mui.com/customization/how-to-customize/#overriding-nested-component-styles)). This property accepts the following subclasses:
+
+| Rule name                 | Description                                             |
+|---------------------------|---------------------------------------------------------|
+| `& .RaLocalesMenuButton-selectedLanguage`      | Applied to the `Typography` component that shows the current language |
```
Comment (WiXSL) [inline docs/LocalesMenuButton.md:74]: Technically, is not a `Typography` is a `div`, mee

**Review by WiXSL — APPROVED** [review]

File: docs/LocalesMenuButton.md (hunk 4 above)
Comment (djhi) [inline docs/LocalesMenuButton.md:74]:
```suggestion
| `& .RaLocalesMenuButton-selectedLanguage`      | Applied to the current language element |
```

File: docs/Reference.md [hunk 5]
```diff
@@ -71,7 +71,7 @@ title: "Reference"
 * [`<ImageInput>`](./ImageInput.md)
 * [`<ImageInputPreview>`](./ImageInput.md#imageinput)
 * [`<Labeled>`](./Labeled.md)
-* [`<LanguageSwitcher>`](https://marmelab.com/ra-enterprise/modules/ra-preferences#languageswitcher-store-the-locale-in-preferences)<img class="icon" src="./img/premium.svg" />
+* [`<LocalesMenuButton>`](./LocalesMenuButton.md)
```
Comment (fzaninotto) [inline docs/Reference.md:74]: It's not ordered alphabetically anymore

**Review by fzaninotto — CHANGES_REQUESTED** [review]
Almost perfect!

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**